tests/installation/test_pip_installer.py

```diff
@@ -62,6 +67,39 @@ def test_requirement_source_type_url():
     assert expected == result
 
 
+@pytest.fixture()
+def poetry(config):
+    poetry = Factory().create_poetry(
+        Path(__file__).parent.parent / "fixtures" / "simple_project"
+    )
+    poetry.set_config(config)
+
+    return poetry
+
+
+@pytest.fixture()
+def manager(poetry):
+    return EnvManager(poetry)
+
+
+@pytest.fixture
+def tmp_venv(tmp_dir, manager):
+    venv_path = Path(tmp_dir) / "venv"
+
+    manager.build_venv(str(venv_path))
+
+    venv = VirtualEnv(venv_path)
+    yield venv
+
+    shutil.rmtree(str(venv.path))
+
+
+def test_can_install_pendulum_sdist(tmp_venv):
+    installer = PipInstaller(tmp_venv, NullIO(), Pool())
+
+    installer.run("install", "--no-binary", ":all:", "pendulum")
```

Comment:
This will fail today because poetry-core `1.0.0a5` cannot be installed from source. When pendulum is attempted to be built using PEP-517.

Using `pip install --no-binary pendulum` should progress further. But will fail when we hit pendulum. Which again attempts to install using setuptools.

This should be resolved once python-poetry/core#24 is merged.